### ISSUE TITLE
Issue with Target Rotation and Radar Echo Consistency in radarsimpy

### ISSUE THREAD

**Issue by Tiffsummerice** [radarsimx/radarsimpy#52]
Dear Dr. Zhengyu Peng,

First of all, I want to express my heartfelt gratitude for sharing your code. It has been incredibly helpful and insightful for my experiments, and I truly appreciate the effort and dedication you’ve put into this work.

I’ve been using your code on a paid GPU in a Windows environment, and while it has greatly benefited my research, I’ve encountered a couple of issues that I’m struggling to resolve. I was hoping you might kindly provide some guidance:

1. When I set the rotation parameter for a target to (0, 0, 90), I noticed that the target does not rotate 90 degrees as expected in the scenario. Could you please clarify why this might be happening?

2. I tested the ball_1m model at different positions and observed that the shape of the radar echo signal for the ball varies at different positions, whereas I expected only a change in magnitude. I’ve been debugging this for quite some time but haven’t been able to determine the cause. Could you please advise on what might be causing this behavior?

Once again, thank you so much for your time and for providing such an excellent resource. I look forward to any insights you might be able to share.

Best regards,
Shengyuan Li

**Comment by Tiffsummerice** [comment]
Hello Dr. Peng, Thank you for taking the time to reply and address my questions. I have prepared the sample code and data, which replicate the issues I mentioned earlier. Please refer to the attached files for details. In the code, I have defined the following targets:
1. Three corner reflectors: The first one reflects signals normally, while the second and third are rotated 90 degrees and 180 degrees around the z-axis, respectively, but they fail to reflect signals properly.
2. Three balls: These are placed at different positions, and their radar echo envelopes show significant differences.
I'm not sure why these are happening, If you need additional details or would like me to perform further tests, please let me know. I sincerely appreciate your guidance and support.
Thank you once again for your help!
[UWBPulseRadar.zip](https://github.com/user-attachments/files/18260277/UWBPulseRadar.zip)
[Ball_echo.zip](https://github.com/user-attachments/files/18260282/Ball_echo.zip)

**Comment by rookiepeng** [comment]
Hi Shengyuan,
If you check the documentation: https://radarsimx.github.io/radarsimpy/sim.html#radarsimpy.simulator.sim_radar
The definition of the rotation is `[yaw, pitch, roll]`, and rotation around z axis is `yaw`.

For the ball, I think this is expected as the ball is not an ideal point target, it is correct to see different envelopes especially in near range. If you want to see the ideal echo, you can use a point target.

**Comment by Tiffsummerice** [comment]
Hi Dr. Peng,
Thank you so much for your clarification and for directing me to the documentation! I followed your guidance, and it resolved the first issue regarding the rotation definition, I really appreciate your help.

As for the second issue, I tested the ultra-wideband radar echoes using an ideal point target, and the results showed excellent consistency across different distances and orientations. Interestingly, I noticed that the echo signals from the ball only exhibit anomalies when positioned very close to the z-axis. When the ball is farther from the z-axis, its ultra-wideband pulse response aligns well with theoretical expectations. I suspect this behavior may be related to the antenna's radiation pattern, and I plan to conduct further experiments to explore this.

Thank you once again for your valuable guidance and support—it has been immensely helpful for my work!

Best regards,
Shengyuan
